packages/flame/test/game/mixins/has_tappables_test.dart

```diff
@@ -0,0 +1,107 @@
+import 'dart:ui';
+
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame/input.dart';
+import 'package:flame_test/flame_test.dart';
+import 'package:test/test.dart';
+
+class _GameWithTappables extends FlameGame with HasTappables {
+  int handledOnTapDown = 0;
+  int handledOnTapUp = 0;
+  int handledOnTapCancel = 0;
+
+  @override
+  bool onTapDown(int pointerId, TapDownInfo info) {
+    final handled = !super.onTapDown(pointerId, info);
+    if (handled) {
+      handledOnTapDown++;
+    }
+    return true;
+  }
+
+  @override
+  bool onTapUp(int pointerId, TapUpInfo info) {
+    final handled = !super.onTapUp(pointerId, info);
+    if (handled) {
+      handledOnTapUp++;
+    }
+    return true;
+  }
+
+  @override
+  bool onTapCancel(int pointerId) {
+    final handled = !super.onTapCancel(pointerId);
```

Comment:
I wonder if we can think on an alternative to always have to check super, can't the engine check if this was handled already on the top of the component tree so the method ins't called and we don't need to always check?